Mon Mar 31 15:07:35 2003	Arnaud Charlet  <charlet@gnat.com>

	* gtk-plug.adb, gtk-socket.adb: Dummy these packages, since they
	are not implemented under Windows.

Tue Jul  9 11:48:58 2002  Arnaud Charlet  <charlet@gnat.com>

	* gdk-window.adb: Remove reference to gdk_window_get_decorations since
	it is undefined under Windows.

	* glib-values.ads: Fix definition of GValue to conform to ms bitfields
	packing under Windows.

*** gdk-window.adb.old	Thu Apr 25 18:07:10 2002
--- gdk-window.adb	Thu Apr 25 18:07:50 2002
***************
*** 118,136 ****
        Decorations : out Gdk_Wm_Decoration;
        Success     : out Boolean)
     is
!       function Internal
!         (Window      : Gdk_Window;
!          Decorations : access Gdk_Wm_Decoration) return Gboolean;
!       pragma Import (C, Internal, "gdk_window_get_decorations");
! 
!       Tmp : aliased Gdk_Wm_Decoration;
! 
     begin
!       Success := To_Boolean (Internal (Window, Tmp'Access));
! 
!       if Success then
!          Decorations := Tmp;
!       end if;
     end Get_Decorations;
  
     ------------------------------
--- 118,128 ----
        Decorations : out Gdk_Wm_Decoration;
        Success     : out Boolean)
     is
!       pragma Warnings (Off, Window);
!       pragma Warnings (Off, Decorations);
!       pragma Warnings (Off, Window);
     begin
!       Success := False;
     end Get_Decorations;
  
     ------------------------------
*** glib-values.ads	4 Sep 2002 14:47:59 -0000	1.13
--- glib-values.ads	8 Nov 2002 11:49:02 -0000
***************
*** 136,142 ****
     --  also provided inside Gtk.Text_Iter.
  
  private
!    type GValue_Data is array (1 .. 2) of Guint64;
     type GValue is record
        g_type : GType := GType_Invalid;
        data   : GValue_Data;
--- 136,143 ----
     --  also provided inside Gtk.Text_Iter.
  
  private
! 
!    type GValue_Data is array (1 .. 20) of Guint8;
     type GValue is record
        g_type : GType := GType_Invalid;
        data   : GValue_Data;
--- gtk-plug.adb	1 Jul 2002 08:07:35 -0000	1.6
+++ gtk-plug.adb	31 Mar 2003 15:06:36 -0000
@@ -27,6 +27,7 @@
 -- executable file  might be covered by the  GNU Public License.     --
 -----------------------------------------------------------------------
 
+pragma Warnings (Off);
 with System;
 
 package body Gtk.Plug is
@@ -48,11 +49,8 @@
    procedure Initialize
      (Plug : access Gtk_Plug_Record'Class; Socket_Id : Guint32)
    is
-      function Internal (Socket_Id : Guint32) return System.Address;
-      pragma Import (C, Internal, "gtk_plug_new");
-
    begin
-      Set_Object (Plug, Internal (Socket_Id));
+      null;
    end Initialize;
 
    ------------
@@ -60,11 +58,8 @@
    ------------
 
    function Get_Id (Plug : access Gtk_Plug_Record) return Guint32 is
-      function Internal (Plug : System.Address) return Guint32;
-      pragma Import (C, Internal, "gtk_plug_get_id");
-
    begin
-      return Internal (Get_Object (Plug));
+      return 0;
    end Get_Id;
 
 end Gtk.Plug;
--- gtk-socket.adb	1 Jul 2002 08:07:35 -0000	1.10
+++ gtk-socket.adb	31 Mar 2003 15:06:36 -0000
@@ -31,6 +31,8 @@
 
 package body Gtk.Socket is
 
+   pragma Warnings (Off);
+
    -------------
    -- Gtk_New --
    -------------
@@ -46,11 +48,8 @@
    ------------------
 
    procedure Initialize (Widget : access Gtk_Socket_Record'Class) is
-      function Internal return System.Address;
-      pragma Import (C, Internal, "gtk_socket_new");
-
    begin
-      Set_Object (Widget, Internal);
+      null;
    end Initialize;
 
    -----------
@@ -58,11 +57,8 @@
    -----------
 
    procedure Steal (Socket : access Gtk_Socket_Record; Wid : Guint32) is
-      procedure Internal (Socket : System.Address; Wid : Guint32);
-      pragma Import (C, Internal, "gtk_socket_steal");
-
    begin
-      Internal (Get_Object (Socket), Wid);
+      null;
    end Steal;
 
    ---------------------
@@ -84,11 +80,8 @@
    ------------
 
    procedure Add_Id (Socket : access Gtk_Socket_Record; Id : Guint32) is
-      procedure Internal (Socket : System.Address; Id : Guint32);
-      pragma Import (C, Internal, "gtk_socket_add_id");
-
    begin
-      Internal (Get_Object (Socket), Id);
+      null;
    end Add_Id;
 
    ------------
@@ -96,11 +89,8 @@
    ------------
 
    function Get_Id (Socket : access Gtk_Socket_Record) return Guint32 is
-      function Internal (Socket : System.Address) return Guint32;
-      pragma Import (C, Internal, "gtk_socket_get_id");
-
    begin
-      return Internal (Get_Object (Socket));
+      return 0;
    end Get_Id;
 
 end Gtk.Socket;
